crates/holochain_state/src/source_chain.rs

```diff
@@ -649,9 +646,30 @@ where
                 WHERE
                 Header.author = :author
                 AND
-                (:range_min IS NULL OR Header.seq >= :range_min)
-                AND
-                (:range_max IS NULL OR Header.seq < :range_max)
+                (:range_start IS NULL AND :range_end IS NULL AND :range_start_hash IS NULL AND :range_end_hash IS NULL AND :range_prior_count IS NULL)
```

Comment:
I think this sql query is already pretty slow but we should actually test this query with a long source chain as it's very likely a full scan of the authored headers with some sub queries that can run for each item. Not the fault of this PR but I think worth a backlog item.